lib/azure/role.rb

```diff
@@ -378,15 +396,17 @@ def setup(params)
                 end
 
                 if params[:tcp_endpoints]
-                  params[:tcp_endpoints].split(',').each do |endpoint|
-                    ports = endpoint.split(':')
-                    if !(ports.length > 1 && ports[1] == params[:port] || ports.length == 1 && ports[0] == params[:port])
+                  params[:tcp_endpoints].split(',').map(&:strip).each do |endpoint|
+                    ports = endpoint.split(':').map(&:strip)
+                    if (ports.length > 1 && !(ports[1] == params[:port])) || (ports.length == 1 && !(ports[0] == params[:port]))
                       xml.InputEndpoint {
                         xml.LocalPort ports[0]
-                        xml.Name 'tcpport_' + ports[0] + '_' + params[:azure_vm_name]
+                        name = "tcpport_#{ports[0]}_#{params[:azure_vm_name]}"
                         if ports.length > 1
+                          xml.Name TCP_ENDPOINTS_MAPPING.keys.include?(ports[1]) ? TCP_ENDPOINTS_MAPPING[ports[1]] : name
```

Comment:
We can have this logic while assigning `name` variable at L#404.
```
name = TCP_ENDPOINTS_MAPPING.keys.include?(ports[1]) ? TCP_ENDPOINTS_MAPPING[ports[1]] : "tcpport_#{ports[0]}_#{params[:azure_vm_name]}"
```

And then assign only the `name` to `xml.Name`.

We should also add specs for this.